charms/katib-db-manager/tests/unit/test_operator.py

```diff
@@ -0,0 +1,136 @@
+    mocked_service_patcher.return_value = lambda x, y: None
+    yield mocked_service_patcher
+
+
+def test_not_leader(
+    harness, mocked_resource_handler, mocked_lightkube_client, mocked_kubernetes_service_patcher
+):
+    """Test that charm waits if not leader."""
+    harness.set_leader(False)
+    harness.begin_with_initial_hooks()
+    harness.container_pebble_ready("katib-db-manager")
+    assert harness.charm.model.unit.status == WaitingStatus("Waiting for leadership")
+
+
+def test_no_relation(
+    harness, mocked_resource_handler, mocked_lightkube_client, mocked_kubernetes_service_patcher
+):
+    """Test no relation scenario."""
+    harness.set_leader(True)
+    harness.add_oci_resource(
+        "oci-image",
+        {
+            "registrypath": "ci-test",
+            "username": "",
+            "password": "",
+        },
+    )
+    harness.begin_with_initial_hooks()
+    harness.container_pebble_ready("katib-db-manager")
+    assert harness.charm.model.unit.status == WaitingStatus(
+        "Waiting for mysql connection information"
+    )
+
+
+def test_mysql_relation(
+    harness, mocked_resource_handler, mocked_lightkube_client, mocked_kubernetes_service_patcher
+):
+    "Test that no error is raised when mysql complete relation exists."
+    harness.begin()
+
+    mysql_app = "mysql_app"
+    mysql_unit = f"{mysql_app}/0"
+
+    rel_id = harness.add_relation("mysql", mysql_app)
+    harness.add_relation_unit(rel_id, mysql_unit)
+
+    # Test complete relation
+    data = {
+        "database": "database",
+        "host": "host",
+        "root_password": "root_password",
+        "port": "port",
+    }
+    harness.update_relation_data(rel_id, mysql_unit, data)
+    with does_not_raise():
+        harness.charm._check_mysql()
+
+
+def test_pebble_layer(
+    harness, mocked_resource_handler, mocked_lightkube_client, mocked_kubernetes_service_patcher
+):
+    """
+    Test creation of Pebble layer given that mysql relation is complete.
+    Only testing specific items.
+    """
+    harness.set_model_name("test_kubeflow")
+    harness.begin_with_initial_hooks()
+    mysql_app = "mysql_app"
+    mysql_unit = f"{mysql_app}/0"
+
+    rel_id = harness.add_relation("mysql", mysql_app)
+    harness.add_relation_unit(rel_id, mysql_unit)
+
+    # Test complete relation
+    data = {
+        "database": "database",
+        "host": "host",
+        "root_password": "root_password",
+        "port": "port",
+    }
+    harness.update_relation_data(rel_id, mysql_unit, data)
+    harness.container_pebble_ready("katib-db-manager")
+    pebble_plan = harness.get_container_pebble_plan("katib-db-manager")
+    assert pebble_plan
+    assert pebble_plan._services
+    pebble_plan_info = pebble_plan.to_dict()
+    assert pebble_plan_info["services"]["katib-db-manager"]["command"] == "./katib-db-manager"
+    test_env = pebble_plan_info["services"]["katib-db-manager"]["environment"]
+    assert 6 == len(test_env)
```

Comment:
Put comment what that `6` means:
`# there should be size (6) environment variables`